apps/site/util/download/archive.tsx

```diff
@@ -0,0 +1,197 @@
+import semVer from 'semver';
+
+import type {
+  DownloadDropdownItem,
+  DownloadKind,
+  DownloadArtifact,
+  OperatingSystem,
+  Platform,
+} from '#site/types';
+import type { NodeRelease } from '#site/types/releases';
+import { OS_NOT_SUPPORTING_INSTALLERS, PLATFORMS } from '#site/util/download';
+import { getNodeDownloadUrl } from '#site/util/url';
+
+import { DIST_URL } from '#site/next.constants';
+
+/**
+ * Checks if a download item is compatible with the given OS, platform, and version.
+ */
+function isCompatible(
+  compatibility: DownloadDropdownItem<Platform>['compatibility'],
+  os: OperatingSystem,
+  platform: Platform,
+  version: string
+): boolean {
+  const {
+    os: osList,
+    platform: platformList,
+    semver: versions,
+  } = compatibility;
+
+  return (
+    (osList?.includes(os) ?? true) &&
+    (platformList?.includes(platform) ?? true) &&
+    (versions?.every(r => semVer.satisfies(version, r)) ?? true)
+  );
+}
+
+type CompatibleArtifactOptions = {
+  platforms?: Record<OperatingSystem, Array<DownloadDropdownItem<Platform>>>;
+  exclude?: Array<string>;
+  version: string;
+  kind?: DownloadKind;
+};
+
+/**
+ * Returns a list of compatible artifacts for the given options.
+ */
+const getCompatibleArtifacts = ({
```

Comment:
How often is this called? Is this memoized?

Reply:
The `buildReleaseArtifacts` function called twice, once for the binary and once for the installer. It’s used in `withDownloadArchive.tsx`, and I don’t think there’s really a need to memoize it  🤔